Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.mm

```diff
@@ -43,10 +45,48 @@
     return [object conformsToProtocol:@protocol(NSSecureCoding)];
 }
 
+static HashSet<String>& typesExemptFromSecureCodingCrash()
+{
+    static NeverDestroyed<HashSet<String>> exemptTypes;
+    return exemptTypes.get();
+}
+
+NO_RETURN static void crashWithClassName(Class objectClass)
+{
+    WebKit::logAndSetCrashLogMessage("NSSecureCoding path used for unexpected object"_s);
+
+    std::array<uint64_t, 6> values { 0, 0, 0, 0, 0, 0 };
+    strncpy(reinterpret_cast<char*>(values.data()), NSStringFromClass(objectClass).UTF8String, sizeof(values));
+    CRASH_WITH_INFO(values[0], values[1], values[2], values[3], values[4], values[5]);
+}
+
 CoreIPCSecureCoding::CoreIPCSecureCoding(id object)
     : m_secureCoding((NSObject<NSSecureCoding> *)object)
 {
     RELEASE_ASSERT(!m_secureCoding || [object conformsToProtocol:@protocol(NSSecureCoding)]);
+
+    static dispatch_once_t onceToken;
+    static bool crashOnSecureCoding { false };
+    dispatch_once(&onceToken, ^{
+        NSArray *array = [[NSUserDefaults standardUserDefaults] arrayForKey:@"WebKitCrashOnSecureCodingKey"];
```

Comment:
Only the UI process should have access to NSUserDefaults.  We're going to have to send a bool in WebProcessCreationParameters, NetworkProcessCreationParameters, GPUProcessCreationParameters, then initialize their process-global value from that bool.

Reply:
*Ack*

Yup!